shared/bytesutil/bytes.go

```diff
@@ -353,5 +355,16 @@ func IsHex(b []byte) (bool, error) {
 	if b == nil {
 		return false, nil
 	}
-	return regexp.Match("^(0x)[0-9a-fA-F]+$", b)
+	return hexRegex.Match(b), nil
+}
+
+// IsHexOfLen checks whether the byte array is a hex number prefixed with '0x' and containing the required number of digits.
+func IsHexOfLen(b []byte, length uint64) (bool, error) {
```

Comment:
looks good! I think we should also remove the `error` from the function signature, since there isn't a path where this can return an error.